authenticator/server/credmgrs/aws_secrets_manager.go

```diff
@@ -11,13 +11,16 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-func newAWSSecretManagerCreds(_ *log.Logger, _ config.Config) (CredentialManager, error) {
+func newAWSSecretManagerCreds(_ *log.Logger, c config.Config) (CredentialManager, error) {
+	if c.AwsRegion == "" {
+		return &awsSecretsManagerCredMgr{}, fmt.Errorf("AWS region not set")
```

Comment:
I'm not too familiar with AWS Secrets Manager, so I have a couple of questions:

- Are AWS Secrets Manager instances always in a specific region, or are they global?
- Do you think a company might have multiple secrets manager instances? 
- If a company had multiple instances of AWS secrets manager, and clients in lots of different regions, how would we make sure the authenticator is looking in the right AWS secrets manager for the client?

Reply:
I wasn't familiar either 😆. 
* I don't see a lot of documentation, but based on the console, I see that secrets in different regions are completely independent. Also, the examples they provide all put the AWS region manually, which makes me feel that yes they are independent in different regions.
* It looks like every Account ID can have its own secrets manager instance, and it is possible to share them with [additional setup](https://aws.amazon.com/premiumsupport/knowledge-center/secrets-manager-share-between-accounts/). I did not dive deep into it, because I viewed it as "advanced usage". However, if you want, we could either add that as a part of this PR or do that in a separate PR.